Add post about VGA text mode

phil-opp opened the pull request:
This is the adaption of the “Printing to Screen” post to the second edition.

phil-opp commented:
Build fails because of https://github.com/rust-lang/rust/issues/48734. Should be fixed with the next nightly.

donald-pinckney commented:
This will give me something to do in airports tomorrow! Should I give feedback via github line comments, or just github comments?

phil-opp commented:
However you like :). Thanks a lot!

donald-pinckney commented on blog/content/second-edition/posts/03-vga-text-buffer/index.md:
```diff
@@ -0,0 +1,605 @@
+
+[memory-mapped I/O]: https://en.wikipedia.org/wiki/Memory-mapped_I/O
+
+Note that memory-mapped hardware might not support all normal RAM operations. For example, a device could only support byte-wise reads and return junk when an `u64` is read. Fortunately, the text buffer supports normal reads and writes (and also bitwise operations), so that we don't have to treat it in special way.
```
Give link / citation that it is true that VGA supports normal reads and writes?

donald-pinckney commented on blog/content/second-edition/posts/03-vga-text-buffer/index.md:
```diff
@@ -0,0 +1,605 @@
+[`'static`]: https://doc.rust-lang.org/book/first-edition/lifetimes.html#static
+
+### Printing
+Now we can use the `Writer` to modify the buffer's characters. First we create a method to write a single ASCII byte (it doesn't compile yet):
```
Actually it does compile

donald-pinckney commented on blog/content/second-edition/posts/03-vga-text-buffer/index.md:
```diff
@@ -0,0 +1,605 @@
+            byte => {
+                ...
+
+                self.buffer().chars[row][col].write(ScreenChar {
```
`.buffer().` should be just `.buffer.`

donald-pinckney commented:
Ok, I just added 3 line comments. The other general comment is: what happened to the blink field??? I was expecting to have the VGA buffer specification match precisely our code representation, but it doesn't, which feels kinda weird.

phil-opp commented:
Thanks a lot for the review! I addressed your comments in 2c2bd0a.

> what happened to the blink field???

My plan was to disable blinking support in the bootloader [through a BIOS call](http://www.ctyme.com/intr/rb-0117.htm). Support for bright backgrounds seems like a more sensible default than support for blinking. I haven't had the time to implement it though.

donald-pinckney commented:
Ok. It's not a big deal either way. As usual it was a great post.
